components/SwipeRow.js

```diff
@@ -144,17 +144,38 @@ class SwipeRow extends Component {
 		// finish up the animation
 		let toValue = 0;
 		if (this._translateX._value >= 0) {
-			// trying to open right
-			if (this._translateX._value > this.props.leftOpenValue * (this.props.swipeToOpenPercent/100)) {
-				// we're more than halfway
-				toValue = this.props.leftOpenValue;
-			}
+			// trying to swipe right
+
+      if (this.swipeInitialX < this._translateX._value) {
+        if (this._translateX._value > this.props.leftOpenValue * (this.props.swipeToOpenPercent/100)) {
+          // we're more than halfway
+          toValue = this.props.leftOpenValue;
+        } else {
+          toValue = 0;
```

Comment:
No need to set `toValue` back to 0 here or below, it's the default value (see line 145)